src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -87,55 +92,56 @@
    watcher :- (schema/protocol Watcher)]
   (let [dir-create? (fn [event]
                       (and (= :create (:type event))
-                           (fs/directory? (:path event))))]
+                           (fs/directory? (:full-path event))))]
     (DirWatchUtils/registerRecursive (:watch-service watcher)
                                      (->> events
                                           (filter dir-create?)
-                                          (map #(.toPath (:path %)))))))
+                                          (map #(.toPath (:full-path %)))))))
 
-(schema/defn retrieve-events :- {WatchKey [WatchEvent]}
+(defn get-event-maps-from-key
+  [watch-key]
+  (let [events (.pollEvents watch-key)]
```

Comment:
My understanding was that `if-let` would still execute if the array was empty. We may get an empty array here, but we should never get, nor return, nil. We could short circuit the mapping if the return value was empty, but I assumed that clojure/jvm would optimize that. I definitely can if you think it's valuable.